src/components/select/Option.svelte

```diff
@@ -0,0 +1,160 @@
+{#if visible}
+  <div
+    aria-selected="{isSelected}"
+    class="{classes}"
+    data-option-value="{value}"
```

Comment:
instead of using the `data-*` attribute and then iterating over the nodes we could just use the `$selectStore` to register each note.

You can see something similar here [#L103](https://github.com/svelte-components/svant/blob/master/src/components/checkbox/Checkbox.svelte#L103)

Reply:
That was originally what I did but I was having reliability issues. This way I am sure that if I can see the option on the screen it has the data properties where I can pull the value and label from. I think with the checkbox there are less edge cases, but here we have a pending option that can become an added option and then we have a search mode that removes options and more complicated things like that. I just think this might be a more reliable way of always having the correct value and label. It also makes it easy to pull from the option later.